### PR TITLE
Return non zero code when errors happen in north

### LINKED ISSUE #42: Make north exit with a non zero exit code on fatal errors
The runtime exits with exit code `0` even in error conditions. Extend the handling of `Event::Error` and `break` with e.g `1`. Change the signature of `main`. Do *not* do a `std::process::exit` in  the main loop when receiving a `Event::Error`.

### PR DESCRIPTION
**Pull request by alfonsoros88**
fixes #42

### CODE REVIEW

File: north/src/main.rs [hunk 1]
```diff
@@ -33,10 +33,27 @@ struct Opt {
...
 }
 
+async fn parse_config(path: &PathBuf) -> Result<Config> {
+    toml::from_str(&read_to_string(path).await?).context(format!(
+        "Failed to read configuration file {}",
+        path.display()
+    ))
+}
+
+fn unwrap_or_exit<T>(result: Result<T>) -> T {
```
Comment (flxo) [inline north/src/main.rs:43]: A very common approach is to have a 
```rust
async run() -> Result<i32> {
    // read the config
   // run the runtime...
   // etc
}
```
Within that you can use `?` as usual and the match on the `Result` with the `exit` appears only once.

File: north/src/main.rs [hunk 2]
```diff
@@ -33,10 +33,27 @@ struct Opt {
     pub debug: bool,
 }
 
+async fn parse_config(path: &PathBuf) -> Result<Config> {
```
Comment (flxo) [inline north/src/main.rs:36]: We can get rid of this one line `fn` with the `run` function mentioned below.

File: north/src/main.rs [hunk 3]
```diff
@@ -33,10 +33,27 @@ struct Opt {
...
+        path.display()
+    ))
+}
+
+fn unwrap_or_exit<T>(result: Result<T>) -> T {
+    match result {
+        Ok(t) => t,
+        Err(err) => {
+            eprintln!("{:?}", err);
+            std::process::exit(1);
```
Comment (flxo) [inline north/src/main.rs:48]: Add a `use std::process`.

File: north/src/main.rs [hunk 4]
```diff
@@ -33,10 +33,27 @@ struct Opt {
...
+        "Failed to read configuration file {}",
+        path.display()
+    ))
+}
+
+fn unwrap_or_exit<T>(result: Result<T>) -> T {
+    match result {
+        Ok(t) => t,
+        Err(err) => {
+            eprintln!("{:?}", err);
```
Comment (flxo) [inline north/src/main.rs:47]: Think the `anyhow::Error` impls `Display` which give nicer output.

File: north/src/main.rs (hunk 2 above)
Comment (alfonsoros88) [reply north/src/main.rs:36]: The config.debug is needed to configure the logging, that's why we have this double check for a possible exit(1).

File: north/src/main.rs (hunk 1 above)
Comment (alfonsoros88) [reply north/src/main.rs:43]: But in this case we need to parse the config to set the logging in main before calling run :(

Comment (flxo) [reply north/src/main.rs:44]: Why? Printing the error from `run` should be done via a direct `eprintln` anyway to be independent from the logger.

Comment (alfonsoros88) [reply north/src/main.rs:44]: because the log filter is set via either opt.debug || config.debug. The latter coming from the parsed config.

**Pull request merged**